cabal-validate/main/Main.hs

```diff
@@ -0,0 +1,953 @@
+module Main where
+
+import Control.Applicative (Alternative (many, (<|>)), (<**>))
+import Control.Exception (Exception (displayException), catch, throw, throwIO)
+import Control.Monad (forM_, unless)
+import Data.Data (Typeable)
+import Data.Map.Strict (Map)
+import qualified Data.Map.Strict as Map
+import Data.Maybe (listToMaybe)
+import Data.Text (Text)
+import qualified Data.Text as T
+import Data.Time.Clock (DiffTime, secondsToDiffTime)
+import Data.Time.Clock.System (getSystemTime, systemToTAITime)
+import Data.Time.Clock.TAI (AbsoluteTime, diffAbsoluteTime)
+import Data.Time.Format (defaultTimeLocale, formatTime)
+import Data.Version (Version, makeVersion, parseVersion, showVersion)
+import GHC.Conc (getNumCapabilities)
+import Options.Applicative
+  ( FlagFields
+  , Mod
+  , Parser
+  , ParserInfo
+  , auto
+  , execParser
+  , flag
+  , flag'
+  , fullDesc
+  , help
+  , helper
+  , hidden
+  , info
+  , long
+  , maybeReader
+  , option
+  , progDesc
+  , short
+  , strOption
+  , switch
+  , value
+  )
+import qualified Options.Applicative as Opt
+import System.Console.ANSI
+  ( Color (Blue, Cyan, Green, Red)
+  , ColorIntensity (Vivid)
+  , ConsoleIntensity (BoldIntensity)
+  , ConsoleLayer (Foreground)
+  , SGR (Reset, SetColor, SetConsoleIntensity)
+  , setSGRCode
+  )
+import System.Console.Terminal.Size (Window (width), size)
+import System.Directory (getCurrentDirectory)
+import System.Exit (ExitCode (ExitFailure, ExitSuccess), exitFailure, exitSuccess)
+import System.Info (arch, fullCompilerVersion, os)
+import Text.ParserCombinators.ReadP (readP_to_S)
+import Turtle (ProcFailed (ProcFailed, procArguments, procCommand, procExitCode), cd, proc, procStrict, when, (</>))
+
+tPutStrLn :: Text -> IO ()
+tPutStrLn = putStrLn . T.unpack
```

Comment:
Why not https://hackage.haskell.org/package/text-2.1.1/docs/Data-Text-IO.html#v:putStrLn?